hack/run-hive-locally.sh

```diff
@@ -0,0 +1,365 @@
+    # Blocks until the existing hive-controllers is updated to have 0 replica
+    oc wait -n "$hive_controllers_ns" --for=jsonpath='{.spec.replicas}'=0 deploy/"$hive_component_controllers"
+    # Blocks until the new hive-clustersync is created and has zero replica
+    while ! oc wait -n "$hive_controllers_ns" --for=jsonpath='{.spec.replicas}'=0 sts/"$hive_component_clustersync"; do
+        sleep 1
+    done
+}
+
+function disable_maintenance_mode {
+    hl_echo "Disabling maintenance mode"
+    oc patch hiveconfig hive -p '{"spec": {"maintenanceMode": false}}' --type merge
+
+    hl_echo "Waiting for the controllers to be scaled up"
+    # Blocks until the existing hive-controllers is updated to have 1 replica
+    while ! oc wait -n "$hive_controllers_ns" --for=jsonpath='{.status.updatedReplicas}'=1 deploy/"$hive_component_controllers"; do
+        sleep 1
+    done
+    # Blocks until the new hive-clustersync is created and has the correct number of replicas
+    while ! oc wait -n "$hive_controllers_ns" --for=jsonpath='{.status.updatedReplicas}'="$hive_clustersync_replicas" sts/"$hive_component_clustersync"; do
+        sleep 1
+    done
+}
+
+function run_hive_operator_locally {
+    hl_echo "Scaling down Deployment/hive-operator"
+    oc scale -n "$hive_operator_ns" deploy/"$hive_component_operator" --replicas=0
+    oc wait -n "$hive_operator_ns" --for=jsonpath='{.spec.replicas}'=0 deploy/"$hive_component_operator"
+
+    hl_echo "Running $component locally, press ctrl + c when you are done"
+    local final_env_values=(
+        HIVE_OPERATOR_NS="$hive_operator_ns" \
+        LOG_LEVEL="$hive_operator_log_level"
+    )
+    env "${final_env_values[@]}" make run-operator 2>&1 | tee "./$artifacts_folder/${hive_component_operator}-logs.txt"
+}
+
+function run_hive_controllers_locally {
+    while true; do
+        # Declare (global) associative arrays
+        declare -gA resolved_envs resolved_envFrom_cms
+
+        # Get in-cluster configurations
+        get_incluster_config
+        resolved_envs["HIVE_NS"]="HIVE_NS=$hive_controllers_ns"
+        resolved_envs["HIVE_CLUSTERSYNC_POD_NAME"]="HIVE_CLUSTERSYNC_POD_NAME=hive-clustersync-0"
+
+        # Enable maintenance mode and wait for the controllers to be scaled down
+        enable_maintenance_mode
+
+        # Run hive-controllers in the background
+        hl_echo "Running $hive_component_controllers locally, press ctrl + c when you are done"
+        # Build the manager binary for the first iteration only
+        if [ $hc_spec_generation -eq 0 ]; then
+            env "${resolved_envs[@]}" LOG_LEVEL="$hive_controllers_log_level" make run 2>&1 |
+            tee "./$artifacts_folder/${hive_component_controllers}-logs-${hc_spec_generation}.txt" &
+        else
+            env "${resolved_envs[@]}" ./bin/manager --log-level="$hive_controllers_log_level" 2>&1 |
+            tee "./$artifacts_folder/${hive_component_controllers}-logs-${hc_spec_generation}.txt" &
+        fi
```

Comment:
Why branch here?

Oh, is it because the `build` dep actually gets rebuilt every time?

Reply:
Yeah, it's a pity that `build` doesn't do the usual `make` thing and only rebuild if sources have changed :(